src/GetStmtHierarchy.cpp

```diff
@@ -0,0 +1,724 @@
+            curr_node_ID = curr_node;
+            e.accept(this);
+        }
+        html << close_node();
+    }
+
+    else if (op->is_interleave()) {
+        html << open_node(op, "interleave_vectors");
+
+        int curr_node = curr_node_ID;
+        for (auto &e : op->vectors) {
+            curr_node_ID = curr_node;
+            e.accept(this);
+        }
+        html << close_node();
+    }
+
+    else if (op->is_extract_element()) {
+        std::vector<Expr> args = op->vectors;
+        args.emplace_back(op->slice_begin());
+        html << open_node(op, "extract_element");
+
+        int curr_node = curr_node_ID;
+        for (auto &e : args) {
+            curr_node_ID = curr_node;
+            e.accept(this);
+        }
+        html << close_node();
+    }
+
+    else if (op->is_slice()) {
+        std::vector<Expr> args = op->vectors;
+        args.emplace_back(op->slice_begin());
+        args.emplace_back(op->slice_stride());
+        args.emplace_back(static_cast<int>(op->indices.size()));
+        html << open_node(op, "slice_vectors");
+
+        int curr_node = curr_node_ID;
+        for (auto &e : args) {
+            curr_node_ID = curr_node;
+            e.accept(this);
+        }
+        html << close_node();
+    }
+
+    else {
+        std::vector<Expr> args = op->vectors;
+        for (int i : op->indices) {
+            args.emplace_back(i);
+        }
+        html << open_node(op, "Shuffle");
+
+        int curr_node = curr_node_ID;
+        for (auto &e : args) {
+            curr_node_ID = curr_node;
+            e.accept(this);
+        }
+        html << close_node();
+    }
+}
+void GetStmtHierarchy::visit(const VectorReduce *op) {
+    html << open_node(op, "vector_reduce");
+
+    int curr_node = curr_node_ID;
+    stringstream op_op;
+    op_op << op->op;
+    html << node_without_children(nullptr, op_op.str());
+
+    curr_node_ID = curr_node;
+    op->value.accept(this);
+
+    html << close_node();
+}
+void GetStmtHierarchy::visit(const Prefetch *op) {
+    internal_error << "\n"
+                   << "GetStmtHierarchy: Prefetch is not supported yet. \n\n";
+}
+void GetStmtHierarchy::visit(const Fork *op) {
+    internal_error << "\n"
+                   << "GetStmtHierarchy: Fork is not supported yet. \n\n";
+}
+void GetStmtHierarchy::visit(const Acquire *op) {
+    html << open_node(op, "acquire");
+
+    int curr_node = curr_node_ID;
+    op->semaphore.accept(this);
+
+    curr_node_ID = curr_node;
+    op->count.accept(this);
+
+    html << close_node();
+}
+void GetStmtHierarchy::visit(const Atomic *op) {
+    if (op->mutex_name.empty()) {
+        html << node_without_children(op, "atomic");
+    } else {
+        html << open_node(op, "atomic");
+        html << node_without_children(nullptr, op->mutex_name);
+        html << close_node();
+    }
+}
+
+string GetStmtHierarchy::generate_stmt_hierarchy_js() {
+    stringstream stmt_hierarchy_js;
+
+    stmt_hierarchy_js
```

Comment:
This is fine as-is, but it might be cleaner to use a C++11 raw string literal instead.